File: ebordo_mobile/lib/models/korisnik.dart
import 'dart:convert';
import 'drzava.dart';

class Korisnik {
  final int korisnikId;
  final String ime;
  final String prezime;
  final String datumRodjenja;
  final String korisnickoIme;
  final Drzava drzavljanstvo;
  final bool isIgrac;
  final bool isTrener;
  final bool isAdmin;
  final bool isAktivan;
  final List<int> slika;

  Korisnik(
      {required this.korisnikId,
      required this.ime,
      required this.prezime,
      required this.datumRodjenja,
      required this.korisnickoIme,
      required this.drzavljanstvo,
      required this.isIgrac,
      required this.isTrener,
      required this.isAdmin,
      required this.isAktivan,
      required this.slika});

  factory Korisnik.fromJson(Map<String, dynamic> json) {
    String stringByte = json["slika"] as String;
    List<int> bytes = base64.decode(stringByte);
    return Korisnik(
        korisnikId: int.parse(json["korisnikId"].toString()),
        ime: json["ime"],
        prezime: json["prezime"],
        datumRodjenja: json["datumRodjenja"],
        korisnickoIme: json["korisnickoIme"],
        drzavljanstvo: Drzava.fromJson(json['drzavljanstvo']),
        isIgrac: json["isIgrac"],
        isTrener: json["isTrener"],
        isAdmin: json["isAdmin"],
        isAktivan: json["isAktivan"],
        slika: bytes);
  }

  Map<String, dynamic> toJson() => {
        "korisnikId": korisnikId,
        "ime": ime,
        "prezime": prezime,
        "datumRodjenja": datumRodjenja,
        "korisnickoIme": korisnickoIme,
        "drzavljanstvo": drzavljanstvo,
        "isIgrac": isIgrac,
        "isTrener": isTrener,
        "isAdmin": isAdmin,
        "isAktivan": isAktivan,
        "slika": slika
      };
}

File: eBordo.Api/Database/Korisnik.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace eBordo.Api.Database
{
    public class Korisnik
    {
        public int korisnikId { get; set; }
        public string ime { get; set; }
        public string prezime { get; set; }
        public DateTime datumRodjenja { get; set; }
        public string adresa { get; set; }
        public string telefon { get; set; }
        public string email{ get; set; }
        public string korisnickoIme { get; set; }
        public string lozinkaHash { get; set; }
        public string lozinkaSalt { get; set; }
        public bool isAktivan { get; set; }

        public Drzava drzavljanstvo { get; set; }
        public int drzavljanstvoId { get; set; }

        public Grad gradRodjenja { get; set; }
        public int gradRodjenjaId { get; set; }

        public Igrac igrac { get; set; }
        public int igracId { get; set; }

        public byte[] Slika { get; set; }

        public Trener trener { get; set; }
        public int trenerID { get; set; }

        public bool isIgrac { get; set; }
        public bool isTrener { get; set; }
        public bool isAdmin { get; set; }
    }
}
